src/cls/IPM/Utils/Module.cls

```diff
@@ -1353,6 +1369,38 @@ ClassMethod LoadNewModule(
     quit tSC
 }
 
+/// Check whether an update from a currently installed module to a new version of the module is a valid operation.
+ClassMethod CheckOperationVersionValidity(
+	ByRef newModule,
+	ByRef params) [ Internal ]
+{
+    set toVersion = newModule.Version
+    set toVersionString = newModule.VersionString
+    set moduleName = newModule.Name
+
+    // Get the currently installed version of the module
+    set moduleCurrent = ##class(%IPM.Storage.Module).NameOpen(moduleName)
+    if moduleCurrent '= "" {
+        set fromVersion = moduleCurrent.Version
+        set fromVersionString = moduleCurrent.VersionString
+    } else {
+        // This means that this is a new install/load of a module. Hence no version from/to validity to check.
+        quit
+    }
+
+    // If the version of the module about to be installed is OLDER than the module that is already installed
+    if fromVersion.Follows(toVersion) {
+        set msg = $$$FormatText("Cannot downgrade %1 from version %2 to %3. To downgrade a module, you must first run the uninstall command. Then, run the install command with the desired module version.",moduleName,fromVersionString,toVersionString)
```

Comment:
Nit: second sentence should say "run the install or load command with the desired module version"